contracts/test/fixture/_pool_tilt.js

```diff
@@ -0,0 +1,142 @@
+
+const { ousdUnits } = require("../helpers");
+
+const { factoryCreatePool } = require("./_pools");
+
+const attackerAddress = "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045";
+let sAttacker;
+let pool;
+
+async function getPoolConfiguration(fixture, pool) {
+  const { rEthBPT, sfrxETHwstETHrEthBPT } = fixture;
+
+  if (!!rEthBPT && pool.address == rEthBPT.address) {
+    return {
+      config: {
+        platform: "balancer",
+        poolType: "metaStable",
+        poolId: await fixture.balancerREthStrategy.balancerPoolId(),
+        pool: rEthBPT,
+        // important for Balancer type pools order of these must match the Balancer pool
+        assetAddressArray: [fixture.reth.address, fixture.weth.address],
+        bptToken: rEthBPT,
+        balancerVault: fixture.balancerVault,
+      },
+    };
+  } else if (
+    !!sfrxETHwstETHrEthBPT &&
+    pool.address == sfrxETHwstETHrEthBPT.address
+  ) {
+    return {
+      config: {
+        platform: "balancer",
+        poolType: "composableStable",
+        poolId: await fixture.balancerSfrxWstRETHStrategy.balancerPoolId(),
+        pool: sfrxETHwstETHrEthBPT,
+        // important for Balancer type pools order of these must match the Balancer pool
+        assetAddressArray: [
+          sfrxETHwstETHrEthBPT.address,
+          addresses.mainnet.wstETH,
+          addresses.mainnet.sfrxETH,
+          addresses.mainnet.rETH,
+        ],
+        bptToken: sfrxETHwstETHrEthBPT,
+        balancerVault: fixture.balancerVault,
+      },
+    };
+  }
+}
+
+async function _fundAttackerOption({
+  asset, // attacking asset
+  assetAmount, // required funded asset amount to perform tilt
+  fixture,
+}) {
+  const attackerEthBalance = await waffle.provider.getBalance(attackerAddress);
+  const assetBalance = await asset.balanceOf(attackerAddress);
+
+  if (
+    attackerEthBalance.gte(ousdUnits("100000")) ||
+    assetBalance.gte(assetAmount)
+  ) {
+    // attacker sufficiently funded
+    return;
+  }
+
+  sAttacker = await impersonateAndFundContract(attackerAddress, "1000000"); // 1m ETH
+
+  if (asset.address == fixture.weth.address) {
+    await mintWETH(fixture.weth, sAttacker, "500000");
+  } else if (asset.address == fixture.sfrxETH.address) {
+    const abi = require("../abi/fraxEthMinter.json");
+    const minter = await ethers.getContractAt(
+      abi,
+      addresses.mainnet.FraxETHMinter
+    );
+    await minter
+      .connect(sAttacker)
+      .submitAndDeposit(attackerAddress, { value: assetAmount });
+  }
+}
+
+/* Generic tilt pool function
+ *
+ */
+async function tiltPool({
+  fixture,
+  /* amount of attacking asset in relation to pool TVL. 100 == 100% of TVL meaning
+   * a pool with 1m TVL will get tilted by 1m of attacking asset.
+   */
+  tiltTvlFactor = 100,
+  attackAsset, // asset used to tilt the pool
+  poolContract,
+}) {
+  const poolConfig = await getPoolConfiguration(fixture, poolContract);
+  pool = await factoryCreatePool(poolConfig.config);
```

Comment:
Using a global variable works, but we will be having issues when we want to parallelize these tests. Also, I feel like there should be cleaner way to do this.

Perhaps have this `tiltPool()` function return a function that'll untilt the pool?? Something like this:
```
function tiltPool() {
    ... 
    return () => untiltPool(pool)
}
```


On the test file:

```
    const untilt = tiltPool()
    ...
    // Do test thing here
    ...
    untilt()
```

Reply:
That's a good point. Have done something similar https://github.com/OriginProtocol/origin-dollar/pull/1849/commits/1c98581fca76f20c039950c7ff070ad730a70595